File: tests/fixtures/header_injection/php/unsafe_set_header.php
<?php
// Unsafe: $_GET['lang'] concatenated into a `header()` line.  The bare
// `header` matcher (exact-match sigil) fires on the call.  Tainted input
// without `\r\n` stripping permits response splitting.
$lang = $_GET['lang'];
header("X-Lang: " . $lang);

File: tests/fixtures/header_injection/rust/unsafe_set_header.rs
// Unsafe: tainted env value flows into `response.headers_mut().insert`.
// HEADER_INJECTION fires on the value argument.
use std::env;

fn handler(response: &mut http::Response<()>) {
    let lang = env::var("LANG").unwrap_or_default();
    let value = http::HeaderValue::from_str(&lang).unwrap();
    response.headers_mut().insert("X-Lang", value);
}
